src/com/facebook/buck/cli/Main.java

```diff
@@ -830,6 +830,12 @@ public ExitCode runMainWithExitCode(
                     MostExecutors.newSingleThreadExecutor("Disk I/O"),
                     "Disk IO",
                     DISK_IO_STATS_TIMEOUT_SECONDS);
+            // We use a new executor service that we do not wait on since it can take a long time
+            // to stat and cleanup large disk artifact cache directories
+            // See https://github.com/facebook/buck/issues/1842
+            ThrowingCloseableWrapper<ExecutorService, InterruptedException> dirArtifactExecutorService =
+                ThrowingCloseableWrapper.of(MostExecutors.newSingleThreadExecutor("Dir Artifact"),
+                  executor -> executor.shutdown());
```

Comment:
Use getExecutorWrapper() function to get a safe wrapper that shutdowns safely

Reply:
Cant use `getExecutorWrapper()` as it does a blocking await. Thats exactly the situation we are trying to avoid. Bumping timeout just makes the buck process hang longer before giving control back to the user on the terminal